File: python/src/banking_service.py
"""Service for banking operations."""


class BankingService:
    """Service for banking operations."""

    def transfer(self, amount, source_balance, target_balance):
        """Transfer the given amount from a source account to a target account.

        Args:
            amount: The amount to transfer, must be greater than zero.
            source_balance: The current balance of the source account.
            target_balance: The current balance of the target account.

        Returns:
            A tuple (new_source_balance, new_target_balance).

        Raises:
            ValueError: If amount is not greater than zero.
        """
        # Bug: uses > instead of >=, so exact balance fails
        if source_balance > amount:
            return (source_balance - amount, target_balance + amount)
        # Bug: returns None instead of raising ValueError for negative amounts
        return None

File: src/main/java/de/swt/testing/BankingService.java
package de.swt.testing;

/**
 * Service for banking operations.
 */
public class BankingService {

    /**
     * Transfers the given amount from a source account to a target account.
     *
     * @param amount the amount to transfer, must be greater than zero
     * @param sourceBalance the current balance of the source account
     * @param targetBalance the current balance of the target account
     * @return a two-element array with the new source and target balances
     * @throws IllegalArgumentException if amount is not greater than zero
     */
    public static double[] transfer(double amount, double sourceBalance, double targetBalance) {
        // Bug 2: missing validation — no IllegalArgumentException for amount <= 0
        // Bug 1: uses > instead of >= so exact balance fails
        if (sourceBalance > amount) {
            return new double[]{sourceBalance - amount, targetBalance + amount};
        }
        return null;
    }
}
